pkg/provider/aws/knowledgebase/eks.go

```diff
@@ -86,9 +103,123 @@ var EksKB = knowledgebase.Build(
+				return fmt.Errorf("no role found for chart %s and source reference %s in HelmChart to S3Bucket expansion", chart.Id(), data.SourceRef.ToId())
+			}
+			dag.AddDependency(role, bucket)
+			return nil
+		},
+		Configure: func(chart *kubernetes.HelmChart, bucket *resources.S3Bucket, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			for _, env := range data.EnvironmentVariables {
+				addEnvVarToChart(chart, bucket, env)
+			}
+			return nil
+		},
+	},
+	knowledgebase.EdgeBuilder[*kubernetes.HelmChart, *resources.RdsInstance]{
+		Configure: func(chart *kubernetes.HelmChart, instance *resources.RdsInstance, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+
+			role := GetRoleForUnit(chart, data.SourceRef)
+			if role == nil {
+				return fmt.Errorf("no role found for chart %s and source reference %s in HelmChart to ddb RdsInstance expansion", chart.Id(), data.SourceRef.ToId())
+			}
+			refs := role.ConstructsRef.Clone()
+			refs.AddAll(instance.ConstructsRef)
+			inlinePol := resources.NewIamInlinePolicy(fmt.Sprintf("%s-connectionpolicy", instance.Name), refs, instance.GetConnectionPolicyDocument())
+			role.InlinePolicies = append(role.InlinePolicies, inlinePol)
+
+			for _, env := range data.EnvironmentVariables {
+				addEnvVarToChart(chart, instance, env)
+			}
+			return nil
+		},
+	},
+	knowledgebase.EdgeBuilder[*kubernetes.HelmChart, *resources.RdsProxy]{
+		Configure: func(chart *kubernetes.HelmChart, proxy *resources.RdsProxy, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			role := GetRoleForUnit(chart, data.SourceRef)
+			if role == nil {
+				return fmt.Errorf("no role found for chart %s and source reference %s in HelmChart to ddb RdsProxy expansion", chart.Id(), data.SourceRef.ToId())
+			}
+			upstreamResources := dag.GetUpstreamResources(proxy)
+			for _, res := range upstreamResources {
+				if tg, ok := res.(*resources.RdsProxyTargetGroup); ok {
+					for _, tgUpstream := range dag.GetDownstreamResources(tg) {
+						if instance, ok := tgUpstream.(*resources.RdsInstance); ok {
+							refs := role.ConstructsRef.Clone()
+							refs.AddAll(instance.ConstructsRef)
+							inlinePol := resources.NewIamInlinePolicy(fmt.Sprintf("%s-connectionpolicy", instance.Name), refs, instance.GetConnectionPolicyDocument())
+							role.InlinePolicies = append(role.InlinePolicies, inlinePol)
+							dag.AddDependency(role, instance)
+						}
+					}
+				}
+			}
+			for _, env := range data.EnvironmentVariables {
+				addEnvVarToChart(chart, proxy, env)
+			}
+			return nil
+		},
+		ValidDestinations: []core.Resource{&resources.RdsInstance{}},
+	},
 )
+
+func GetRoleForUnit(chart *kubernetes.HelmChart, ref core.AnnotationKey) *resources.IamRole {
+	for _, val := range chart.Values {
+		if iacVal, ok := val.(core.IaCValue); ok {
+			if role, ok := iacVal.Resource.(*resources.IamRole); ok {
+				singleRef, ok := role.ConstructsRef.GetSingle()
+				if ok && singleRef == ref {
+					return role
+				}
```

Comment:
here we also want to make sure that the role has a single ref i believe otherwise were not positive that its the right role, because something like the node group role could have many refs

Reply:
you mean something like 
```
func GetRoleForUnit(chart *kubernetes.HelmChart, ref core.AnnotationKey) *resources.IamRole {
	for key, val := range chart.Values {
		if kubernetes.GenerateRoleArnPlaceholder(ref.ID) == key {
			if iacVal, ok := val.(core.IaCValue); ok {
				if role, ok := iacVal.Resource.(*resources.IamRole); ok {
					return role
				}
			}
		}
	}
	return nil
}
```

Reply:
Yeah, something like that.